pkg/dao/schema/resource_component.go

```diff
@@ -84,6 +84,13 @@ func (ResourceComponent) Fields() []ent.Field {
 			Annotations(
 				entx.SkipInput(),
 				entx.SkipStoringField()),
+		field.String("index_key").
+			Comment("Index key to identify the component instance.").
```

Comment:
This identifier is used to match a unique Terraform resource, since `ResourceComponents` utilize the Terraform resource's attribute `id` as the `name`. However, an issue arises when planning to create or alter a resource, as these attributes id may change. To ensure that changes made by Terraform correspond to the correct ResourceComponents, we need to record this data.